### PR TITLE
[56004] [Issue 57] - Automatic update of certificate / identity server mock to test functionality 

### PR DESCRIPTION
**Pull request by ant3k96**
Adding implementation and test regarding issue 57. Verification token previously hard coded in _OAuthConf.cs_ will be downloaded automatically from identity server with a use of method **GetPublicKey**.  To test method **GetPublicKey** located in _AuthenticationManager.cs_ I created identity server mock as a separate project in the solution.

Test method **Get_EndpointReturnSuccessAndCorrectContentType** in _IdentityServerTest.cs_ checks if the public key downloaded from the server mock matches public key obtained in test method. Both are created from the same certificate. Public key located in test method is processed from certificate with a use of **Chilkat** package.

### CODE REVIEW

File: NDB.Covid19/NDB.Covid19.Test/Tests/OAuth2/IdentityServerTest.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            jks.ToJwkSet(password, sbJwkSet);
+            Chilkat.JsonObject jwkSet = new Chilkat.JsonObject();
+            jwkSet.LoadSb(sbJwkSet);
+            jwkSet.EmitCompact = false;
+            var jwksCheck = jwkSet.Emit();
+            JsonWebKeySet jwkscheck = new JsonWebKeySet(jwksCheck);
+            List<JsonWebKey> keyList2 = new List<JsonWebKey>(jwkscheck.Keys);
+            string publicKey2 = keyList2[0].X5c[0];
+            //response from server
+            var response = await authenticationManager.client.GetAsync(url);
```
Comment (nc-kano) [inline NDB.Covid19/NDB.Covid19.Test/Tests/OAuth2/IdentityServerTest.cs:56]: use explicit types

File: NDB.Covid19/IdentityServerMock/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+        }
+        public void AddOldKey(IServiceCollection services)
+        {
+            var builder = services.AddIdentityServerBuilder();
+            builder.AddValidationKey(OldCertificate);
+        }
+        public void AddKey(IServiceCollection services)
+        {
+            var builder = services.AddIdentityServerBuilder();
```
Comment (nc-kano) [inline NDB.Covid19/IdentityServerMock/Startup.cs:114]: do not use wars in whole file

**Review by nc-kano — COMMENTED** [review]
Looks more or less ok for me. just replace vars with explicit types and I will approve

File: NDB.Covid19/IdentityServerMock/Startup.cs (hunk 2 above)
Comment (ant3k96) [reply NDB.Covid19/IdentityServerMock/Startup.cs:114]: Ok will do that

**Review by Michael-Olesen — APPROVED** [review]

**Pull request merged**